src/frontend/markdown.rs

```diff
@@ -181,6 +187,189 @@ impl InlineRule for MathEquationScanner {
     }
 }
 
+// TOC SECTION
+
+#[derive(Debug, Default, Clone)]
+pub struct Toc {
+    entries: Vec<TocEntry>,
+}
+
+impl Toc {
+    fn count_entries_with_level(&self, level: u8) -> usize {
+        self.entries.iter().filter(|e| e.level == level).count()
+    }
+
+    fn print_entry(&self, fmt: &mut dyn Renderer) {
+        fmt.open("ul", &[]);
+        self.entries.iter().for_each(|entry| {
+            fmt.open("li", &[]);
+            fmt.open("a", &[("href", "#".to_owned() + &entry.id)]);
+
+            fmt.text(&format!("{} {}", entry.sec_number, entry.name));
+
+            fmt.close("a");
+
+            entry.children.print_entry(fmt);
+
+            fmt.close("li");
+        });
+
+        fmt.close("ul");
+    }
+}
+
+impl NodeValue for Toc {
+    fn render(&self, node: &Node, fmt: &mut dyn Renderer) {
+        let mut attrs = node.attrs.clone();
+
+        attrs.push((
+            "class",
+            "not-prose float-left mr-20 w-80 menu rounded-box".into(),
```

Comment:
With this you get text flowing to the right of the toc, which doesnt look so good in my opinion. I would make another div around it with `w-full` to prevent that.